### PR TITLE
fix!: Expression parameter strings are now delimited by a comma. `get_expression_parameter_string` and `get_string_parameter_string` have been removed from the instruction module.

### PR DESCRIPTION
**Pull request by MarquessV**
List of expressions were being printed without a delimmiting comma. This fixes that issue, as well as refactors some related methods to not allocate in Display.

### CODE REVIEW

File: quil-rs/src/instruction/mod.rs [hunk 1]
```diff
@@ -233,26 +240,48 @@ fn write_qubit_parameters(f: &mut fmt::Formatter, qubits: &[Qubit]) -> fmt::Resu
...
+) -> fmt::Result {
+    if values.is_empty() {
+        return Ok(());
+    }
+
+    write!(f, "{}", values[0])?;
+    for value in values[1..].iter() {
+        write!(f, ", {value}")?;
+    }
+    Ok(())
```
Comment (Shadow53) [inline quil-rs/src/instruction/mod.rs:255]: This works fine as-is, but stylistically could be written better. My preference is usually

```suggestion
    let mut iter = values.iter();
    
    if let Some(value) = iter.next() {
        write!(f, "{value}")?;
    }
    
    for value in iter {
        write!(f, ", {value}")?;
    }
    
    Ok(())
```

But I'm also okay with

```suggestion
    for (i, value) in values.iter().enumerate() {
        if i != 0 {
            write!(f, ", ")?;
        }
        write!(f, "{value}")?;
    }
    Ok(())
```

File: quil-rs/src/instruction/mod.rs [hunk 2]
```diff
@@ -233,26 +240,48 @@ fn write_qubit_parameters(f: &mut fmt::Formatter, qubits: &[Qubit]) -> fmt::Resu
...
-        .map(|param| format!("%{param}"))
-        .collect::<Vec<_>>()
-        .join(",");
-    format!("({parameter_str})")
+    write!(f, "(")?;
+    write!(f, "%{}", parameters[0])?;
+    for parameter in parameters[1..].iter() {
+        write!(f, ", %{parameter}")?;
+    }
+    write!(f, ")")
```
[the remark this answers is not in the document]
Comment (Shadow53) [reply quil-rs/src/instruction/mod.rs:284]: If not, see suggestions on `write_comma_separated_list`

**Review by Shadow53 — CHANGES_REQUESTED** [review]

File: quil-rs/src/instruction/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarquessV) [reply quil-rs/src/instruction/mod.rs:286]: I think you called out the right tradeoff, and I'm not sure which is best. My initial hesitation with a prefix argument is that all of this is part of the public API, but I think as long as we restrict it, it's not going to be binding. Coming back to this fresh I don't think any of these write methods really belong in the public API. I don't think they are being used outside of the module, and if they are, I think we can revisit what a proper instruction formatting API might look like. 

All that to say, I ultimately went with the prefix argument.

**Review by Shadow53 — APPROVED** [review]
LGTM

**Pull request merged**